Expose additional values for prometheus configmap-reload in values.yaml

antonosmond opened the pull request:
@mgoodness 

**What this PR does / why we need it**:
The config-reload image used in the prometheus chart allows multiple directories to be monitored to trigger config-reloads so expose `extraArgs` and `extraConfigMapMounts` in values.yaml so this functionality can be used.

This allows us to do things like separating the alert configs from the main prometheus config making them easier to manage and update.

Commit:
Add extraArgs and extraConfigmapMounts for prometheus configmapReload container to support monitoring additional volumes to trigger config reloads

Commit:
Incremented chart version

antonosmond commented:
/assign @mgoodness

timstoop commented on stable/prometheus/values.yaml:
```diff
@@ -184,6 +184,19 @@ configmapReload:
+  extraConfigmapMounts: []
+    # - name: prometheus-alerts
+    #   mountPath: /etc/alerts.d
+    #   configMap: prometheus-alerts
```
I may not fully understand how the code works, but it looks like this line from the example is not being parsed by the actual server-deployment.yaml, correct?

antonosmond replied:
Good spot you're absolutely correct! I've missed the bit under volumes e.g. [here](https://github.com/echo-health/charts/blob/master/stable/prometheus/templates/server-deployment.yaml#L148-L152). I'll add that in.

timstoop reviewed (changes requested):
As you're adding new features, please bump the minor version of Chart.

Commit:
Add missing volume mapping for configmapReload extraConfigmapMounts in prometheus chart

antonosmond committed:
Merge branch 'master' into master

antonosmond commented:
@timstoop - made the suggested changes. Thanks!

Commit:
Added appVersion as now required by CircleCI

rtluckie committed:
[stable/minio] Fix configmap issue, add ingress resource and general refactor (#4281)

* Rename manifests to align with best practices

* Refactor minio chart

- add ingress resource
- consolidate svc resource to support all deployment modes
- update labels and selectors to align with helm best practices
- general cleanup to align with helm best practices/patterns observed in `helm create`
- update values, README and _helpers accordingly
- bump image tag
- bump chart version

* Fix issue caused by ConfigMaps now being mounted ReadOnly

Tested on:
k8s 1.8.10 and 1.9.6

Related:
https://github.com/kubernetes/kubernetes/pull/58720

Fixes:
https://github.com/kubernetes/charts/issues/4272

* Bump chart version to 1.0.0

timstoop commented:
/ok-to-test

timstoop commented:
/lgtm

timstoop commented:
/assign @mgoodness

jpoon committed:
[stable/ethereum] Add Chart for Deploying Private Ethereum Network (#3220)

* add chart for deploying private ethereum network. closes #2088.

* fix: update readme. add privatekey for default account to fund

* fix: add private key to values. import-geth-account to geth nodes

* fix: decrease difficulty of network. increase gas limit, allow remote connections to geth-tx

* fix: make difficulty/gasLimit configurable. use versioned ethstats image

* fix: use component labels

* fix: remove LICENSE

* fix: as per code review, remove keys

* docs: add documentation on creating a new geth account

* fix: clarify documentation on creating a new account

* add custom test values.yaml

* move chart to stable

* use latest _helpers

antonosmond commented:
@timstoop @mgoodness Hold off doing anything with this. I need to split the volumes from the mounts as currently, if for example prometheus-server AND configmap-reload both want to mount the same extraConfigmapMounts, it'll try to create the volume twice and fail. This is because it's creating the volumes based on the mounts [here](https://github.com/echo-health/charts/blob/e15cf015e88c68b2b5bf1e33b1dabd8043e399b5/stable/prometheus/templates/server-deployment.yaml#L148-L152)

arbourd committed:
Fix Vault for `cert` and `token` auth (#4425)

* Fix Vault for `cert` and `token` auth

- `caCert` was renamed to `useCaCert` but not in README or values.yaml
- `vault-client-key` and `vault-client-cert` should only need to be set
  when the auth mode has been set to `cert`. Without the check, `approle`
  and `token` auth will fail (due to Go not being able to parse the PEM
  of "")

* Bump minor

AdamDang committed:
Update stable/concourse/README.md (#4488)

* Add version for concourse

Add version for concourse

* Update README.md

Delete the duplicated word "Volume volume"

Chili-Man committed:
[stable/sumologic-fluentd] Support creating persistence directory (#3453)

* Add support for creating the host path

* remove unnecessary else

* update documentation to include note about minimum k8s version for certain params

* Remove the Daemonset when the Sumologic collector URL value is not provided

* undo

Commit:
[stable/drupal] Release 0.11.13 (#4500)

* drupal: update to `bitnami/drupal:8.5.1-r0`

* drupal: bump chart appVersion to `8.5.1`

* drupal: bump chart version to `0.11.13`

AdamDang committed:
Update stable/mongodb/README.md (#4497)

* Add version for mongodb

Add version for mongodb

* Update stable/mongodb/README.md

Update stable/mongodb/README.md

AdamDang committed:
Update stable/quassel/README.md (#4494)

* Add version for quasse

Add version for quasse

* Update stable/quassel/README.md

Update stable/quassel/README.md

AdamDang committed:
Update stable/mariadb/README.md (#4496)

* Add version for mariadb

Add version for mariadb

* Update stable/mariadb/README.md

Update stable/mariadb/README.md

AdamDang committed:
Update /stable/rabbitmq/README.md (#4493)

* Add version for rabbitmq

Add version for rabbitmq

* Update /stable/rabbitmq/README.md

Update /stable/rabbitmq/README.md

Chr1st0ph committed:
stable/selenium values.yaml: fixed comments (#4483)

* stable/selenium values.yaml: fixed comments

* stable/selenium update Chart.yaml version

AdamDang committed:
Update stable/redis/README.md (#4489)

* Add version for redis

Add version for redis

* Update README.md

Delete duplicated word.

AdamDang committed:
Update stable/postgresql/README.md (#4495)

* Add version for postgresql

Add version for postgresql

* Update stable/postgresql/README.md

Update stable/postgresql/README.md

* Add appVersion key

Add appVersion key

* Modify appVersion

Modify appVersion

AdamDang committed:
Update kapacitor/README.md (#4492)

* Add version for kapacitor

Add version for kapacitor

* Update kapacitor/README.md

Update kapacitor/README.md

* Add appVersion key

Add appVersion key

* Modify appVersion

Modify appVersion

* Update Chart.yaml

AdamDang committed:
Update stable/influxdb/README.md (#4491)

* Add version for influxdb

Add version for influxdb

* Update stable/influxdb/README.md

Update stable/influxdb/README.md

* Add appVersion key

Add appVersion key

* Modify appVersion

Modify appVersion

* Update Chart.yaml

AdamDang committed:
Update stable/chronograf/README.md (#4490)

* Add version for chronograf

Add version for chronograf

* Update stable/chronograf/README.md

Delete duplicated word.

* Add appVersion key

Add appVersion key

* Modify appVersion

Modify appVersion

* Update Chart.yaml

cimomo committed:
concourse: Small fixes in README and templates (#4484)

* concourse: Small fixes in README and templates

* Update chart version (version got bumped since the earlier PR)

Commit:
[stable/prometheus] refactor extraArgs and extraConfigmapMounts to work around volume issue

antonosmond commented:
@mgoodness @timstoop closing in favour of https://github.com/kubernetes/charts/pull/4511
